cmake: Apply cmake-format

jdemel opened the pull request:
This commit adds `.cmake-format.py`. It is mostly inspired by the GNU Radio version of it.

Steps to reproduce these results:
- `pip3 install cmakelang`
- `cmake-format --dump-config python > .cmake-format.py`
- Modify .cmake-format.py to be close to the GNU Radio version.
- `cmake-format -c .cmake-format.py -i $(git ls-files | grep CMakeLists.txt) $(git ls-files | grep .cmake)`

Fixes #600
Fixes #423

marcusmueller commented on cmake/Toolchains/aarch64-linux-gnu.cmake:
```diff
@@ -9,7 +9,10 @@
+if(
+  MINGW
+  OR CYGWIN
+  OR WIN32)
```
this is extremely ugly.

jdemel replied:
I re-added the corresponding file from gnuradio and the result is almost the same but without the new line after `if(`. I'd assume this is intended.

marcusmueller commented on cmake/Toolchains/arm_cortex_a72_hardfp_native.cmake:
```diff
@@ -11,7 +11,16 @@
+set(
+  CMAKE_C_FLAGS
+  ${CMAKE_CXX_FLAGS}
+  CACHE STRING "" FORCE) #same flags for C sources
```
Not sure, but this looks pretty different than GR's formatting. You're closing an issue that says "format like GNU Radio" with this!

marcusmueller commented:
If you want to be close to the GNU Radio version, so go and copy the GNU Radio version, I'd say.

jdemel commented:
> If you want to be close to the GNU Radio version, so go and copy the GNU Radio version, I'd say.

I'm pretty sure, I did. I saw you @marcusmueller made some further adjustments to the GR CMake formatting. I should probably get an updated version of this.

jdemel committed:
cmake: Add .cmake-format.py

We want to use this to auto-format our cmake files.

Signed-off-by: Johannes Demel <demel@uni-bremen.de>

jdemel committed:
cmake: Apply .cmake-format.py

Run:
```
cmake-format -c .cmake-format.py -i $(git ls-files | grep CMakeLists.txt) $(git ls-files cmake/ | grep .cmake)
```
This applies the cmake formatting as defined in all files.

In order to install `cmake-format`, you might want to run
`pip install cmakelang`
Or use another package manager of your choice.

Signed-off-by: Johannes Demel <demel@uni-bremen.de>

jdemel commented:
I copied the `.clang-format.py` file from GNU Radio again. I expect that the result looks like what GR does now. I diff'd the files and the VOLK version of `.clang-format.py` is the same + some comments that drop out of the default. I'd argue this is nice for future reference.

marcusmueller reviewed (approved):
looks good; thanks!
